compiler/rustc_typeck/src/astconv/mod.rs

```diff
@@ -1145,13 +1140,19 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
 
         // We have already adjusted the item name above, so compare with `ident.normalize_to_macros_2_0()` instead
         // of calling `filter_by_name_and_kind`.
-        let assoc_item = tcx
-            .associated_items(candidate.def_id())
-            .filter_by_name_unhygienic(assoc_ident.name)
+        let filtered_assoc_items =
+            || tcx.associated_items(candidate.def_id()).filter_by_name_unhygienic(assoc_ident.name);
```

Comment:
pass in the kind you're looking for as an argument and include the `find` part in the closure, too.